pallets/polkadex-ido/src/lib.rs

```diff
@@ -125,1451 +63,628 @@ type BalanceOf<T> =
-		StartBlockMustBeLessThanEndblock,
-		/// Start `start_block` is less than `voting_period`  error
-		StartBlockMustBeGreaterThanVotingPeriod,
-		/// Round already approved  error
-		RoundAlreadyApproved,
-		/// Attempt to redraw already redrawn raise amount
-		FundRaisedRedrawn,
-		/// invalid period
-		PeriodError,
-		/// attempting to vote for a round already ended voting period error
-		VotingEnded,
-		/// Attempt to vote more than once error
-		DuplicateVote,
-		/// Failed to move investor funds to reserve balance error
-		FailedToMoveBalanceToReserve,
-		/// Funding round not approved error
-		FundingRoundNotApproved,
-		/// CID bytes exceeding expected size error
-		CidReachedMaxSize,
-		/// `vesting_per_block` invalid error
-		VestingPerBlockMustGreaterThanZero,
-		/// Attempt to min native token in ido round creation error
-		MintNativeTokenForbidden,
-		/// Attempt to vote with 0 amount error
-		VoteCannotBeZero,
-		/// Investor attempting unlock already unlocked funds error
-		AlreadyUnlockedInvestorRegistrationFunds,
-		/// Investor attempting unlock investor registration fund when lock period is no expired
-		/// yet
-		UnlockedInvestorRegistrationFundBlocked,
-		/// Insufficient balance in an account
-		InsufficientBalance,
-	}
+    use frame_support::{
+        pallet_prelude::*,
+        traits::tokens::fungibles::{Create, Inspect, Mutate},
+        PalletId,
+    };
+    use frame_system::{offchain::CreateSignedTransaction, pallet_prelude::*};
+    use sp_core::{H160, H256};
+    use sp_runtime::traits::{One, CheckedMul, CheckedDiv};
+    use sp_std::prelude::*;
+    use sp_runtime::{FixedU128, FixedPointNumber};
+
+    use super::*;
+
+
+    use polkadex_primitives::assets::AssetId;
+    use pallet_polkadex_ido_primitives::AccountId;
+
+    /// The module configuration trait.
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// The origin which may attests the investor to take part in the IDO pallet.
+        type GovernanceOrigin: EnsureOrigin<Self::Origin, Success=Self::AccountId>;
+        /// The treasury mechanism.
+        #[pallet::constant]
+        type TreasuryAccountId: Get<Self::AccountId>;
+        /// Balances Pallet
+        type Currency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId> + fungible::Inspect<Self::AccountId>;
+        /// The basic amount of funds that must be reserved for an Polkadex.
+        #[pallet::constant]
+        type IDOPDXAmount: Get<BalanceOf<Self>>;
+        /// Maximum supply for IDO
+        #[pallet::constant]
+        type MaxSupply: Get<BalanceOf<Self>>;
+        /// The generator used to supply randomness to IDO
+        type Randomness: Randomness<Self::Hash, Self::BlockNumber>;
+        /// Randomness Source for random participant seed
+        type RandomnessSource: Randomness<H256, Self::BlockNumber>;
```

Comment:
RandomnessSource and Randomness are identical., remove one of it.

Reply:
Removed RandomnessSource ✅